qiita_db/test/test_data.py

```diff
@@ -478,6 +478,38 @@ def test_create_data_type_only(self):
         # preprocessed_data_id, filepath_id
         self.assertEqual(obs, [[3, obs_id - 1], [3, obs_id]])
 
+    def test_delete(self):
+        """Correctly deletes a preprocessed data"""
+        # testing regular delete
+        ppd = PreprocessedData.create(
+            self.study, self.params_table,
+            self.params_id, self.filepaths, prep_template=self.prep_template,
+            ebi_submission_accession=self.ebi_submission_accession,
+            ebi_study_accession=self.ebi_study_accession)
+        PreprocessedData.delete(ppd.id)
```

Comment:
@ElDeveloper any thoughts on this? I'm fine with the code as it is. The reasons why I do not think that more checks are needed are:
 - Everything is in a queue. If one statement fails, all of them fail.
 - If one of the other tables are not removed, SQL will complain due to FK restriction.

I was just proposing an alternative solution, to see if @antgonza and @squirrelo agree on something intermediate. @squirrelo given these arguments, are you still feeling strong on put some extra checks in place?